<a href="https://colab.research.google.com/github/AshikSathiya/Project2/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project 2

---



In [2]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


#First choice: Body Fat Prediction Dataset


In [3]:
# load the dataset
fpath = '/content/drive/MyDrive/CodingDojo/Project 2/bodyfat.csv'
df = pd.read_csv(fpath, header=None)

df.info()
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253 entries, 0 to 252
Data columns (total 15 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       253 non-null    object
 1   1       253 non-null    object
 2   2       253 non-null    object
 3   3       253 non-null    object
 4   4       253 non-null    object
 5   5       253 non-null    object
 6   6       253 non-null    object
 7   7       253 non-null    object
 8   8       253 non-null    object
 9   9       253 non-null    object
 10  10      253 non-null    object
 11  11      253 non-null    object
 12  12      253 non-null    object
 13  13      253 non-null    object
 14  14      253 non-null    object
dtypes: object(15)
memory usage: 29.8+ KB


0        1    2       3       4     5      6        7      8      9   \
0  Density  BodyFat  Age  Weight  Height  Neck  Chest  Abdomen    Hip  Thigh   
1   1.0708     12.3   23  154.25   67.75  36.2   93.1     85.2   94.5   59.0   
2   1.0853      6.1   22  173.25   72.25  38.5   93.6     83.0   98.7   58.7   
3   1.0414     25.3   22  154.00   66.25  34.0   95.8     87.9   99.2   59.6   
4   1.0751     10.4   26  184.75   72.25  37.4  101.8     86.4  101.2   60.1   

     10     11      12       13     14  
0  Knee  Ankle  Biceps  Forearm  Wrist  
1  37.3   21.9    32.0     27.4   17.1  
2  37.3   23.4    30.5     28.9   18.2  
3  38.9   24.0    28.8     25.2   16.6  
4  37.3   22.8    32.4     29.4   18.2

##1. Source of data


This dataset comes from Kaggle

##2. Brief description of data



The dataset contains estimates of the percentage of body fat in 252 men based on underwater weighing and various body circumference measurements. It is suitable for educational purposes to demonstrate multiple regression techniques, aiming to find easier and more cost-effective methods to estimate body fat accurately, as direct measurement through underwater weighing can be inconvenient and costly.






##3. What is the target?



Body Fat

##4. What does one row represent? (A person? A business? An event? A product?)



One row represents a man and his measuremnts.

##5. Is this a classification or regression problem?



This is a regression model as we are trying to predict the body fat percentage based on features

##6. How many features does the data have?



14 features

##7. How many rows are in the dataset?



252 rows

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?



The challenges primarily lie on the modeling portion of this project. The data set is relatively very clean. Something strange is that all the columns are object datatypes despite the instances being primarily numeric which is convenient for regression. I will have to convert those datatypes from object to numeric. There is also so many features which is really nice. It will be fun to explore the datset and see what we cna find or visualise during the data exploration process. Something I am hoping to work on is my ability to better tune my models. I am hoping this project gives me a chance to do just that.

#Second choice: Student Mental Health



In [6]:
# load the dataset
fpath1 = '/content/drive/MyDrive/CodingDojo/Project 2/Student Mental health.csv'
df1 = pd.read_csv(fpath1, header=None)

df1.info()
df1.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102 entries, 0 to 101
Data columns (total 11 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   0       102 non-null    object
 1   1       102 non-null    object
 2   2       101 non-null    object
 3   3       102 non-null    object
 4   4       102 non-null    object
 5   5       102 non-null    object
 6   6       102 non-null    object
 7   7       102 non-null    object
 8   8       102 non-null    object
 9   9       102 non-null    object
 10  10      102 non-null    object
dtypes: object(11)
memory usage: 8.9+ KB


0                   1    2                     3   \
0       Timestamp  Choose your gender  Age  What is your course?   
1  8/7/2020 12:02              Female   18           Engineering   
2  8/7/2020 12:04                Male   21     Islamic education   
3  8/7/2020 12:05                Male   19                   BIT   
4  8/7/2020 12:06              Female   22                  Laws   

                           4                   5               6   \
0  Your current year of Study  What is your CGPA?  Marital status   
1                      year 1         3.00 - 3.49              No   
2                      year 2         3.00 - 3.49              No   
3                      Year 1         3.00 - 3.49              No   
4                      year 3         3.00 - 3.49             Yes   

                        7                     8                          9   \
0  Do you have Depression?  Do you have Anxiety?  Do you have Panic attack?   
1                      Yes                    No                        Yes   
2                       No                   Yes                         No   
3                      Yes                   Yes                        Yes   
4                      Yes                    No                         No   

                                             10  
0  Did you seek any specialist for a treatment?  
1                                            No  
2                                            No  
3                                            No  
4                                            No

## New Section

##1. Source of data


Kaggle

##2. Brief description of data



This statistical research focuses on investigating the potential effects of mental health on students' Cumulative Grade Point Average (CGPA). The dataset was gathered through a survey conducted using Google Forms, targeting university students. The primary objective of the survey was to gain insights into the students' current academic situation, along with exploring the various aspects of their mental health. By analyzing this dataset, researchers aim to understand any potential correlations between mental health indicators and academic performance, seeking to shed light on the impact of students' well-being on their CGPA. This study holds significance as it could provide valuable information to educators, institutions, and policymakers in designing interventions and support systems to foster both academic excellence and mental well-being among students.






##3. What is the target?



Do you have Depression

##4. What does one row represent? (A person? A business? An event? A product?)



Student

##5. Is this a classification or regression problem?



Classification

##6. How many features does the data have?



9

##7. How many rows are in the dataset?



102 rows

##8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?



There are a couple challenges I foresee with this dataset. Firstly the column names are all very long and complicated and must be changed to something simpler. Next is that someobject datatypes shouldbe converted to numeric. The biggest challenge I foresee lies in the "What is your CGPA?". The values within this column are in a range. An example of an entry is "3.00 - 3.49". Figuring how to use this is going to be interesting. Overall very excited about this dataset.